lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -913,7 +921,28 @@ static void lookupVisibleDynamicMemberLookupDecls(
     Type baseType, KeyPathDynamicMemberConsumer &consumer,
     const DeclContext *dc, LookupState LS, DeclVisibilityKind reason,
     LazyResolver *typeResolver, GenericSignatureBuilder *GSB,
-    VisitedSet &visited) {
+    VisitedSet &visited, llvm::DenseSet<TypeBase *> &seenDynamicLookup);
+
+static void lookupVisibleMemberAndDynamicMemberDecls(
```

Comment:
Function names are getting confusing. Could you add doc-comments to added functions in this PR and #24520 ?

Reply:
This function is so we have a place to introduce recursion for dynamic lookup.  I wanted to put all the calls in `lookupVisibleMemberDeclsImpl`, but that is already recursive in the wrong way for doing dynamic member lookup :-(  I'll think about this some more and see if I can come up with better names.  Will add doc comments as requested too.